policykit/integrations/discourse/tasks.py

```diff
@@ -15,21 +15,25 @@
 
 logger = logging.getLogger(__name__)
 
-def is_policykit_action(community, call_type, topic, username):
-    if username == 'PolicyKit': ## TODO: Compare IDs in future, not usernames
-        return True
-
-    current_time_minus = datetime.datetime.now() - datetime.timedelta(minutes=2)
-    logs = LogAPICall.objects.filter(
-        proposal_time__gte=current_time_minus,
-        call_type=call_type
-    )
-    if logs.exists():
-        for log in logs:
-            j_info = json.loads(log.extra_info)
-            if topic['id'] == j_info['id']:
-                return True
-    return False
+def should_create_action(community, call_type, topic, username):
+    # If topic already has an object, don't create a new object for it.
+    if DiscourseCreateTopic.objects.filter(id=topic['id']).exists():
+        return False
+
+    created_at = topic['created_at']
+    created_at = created_at.replace("Z", "+00:00")
+    created_at = datetime.datetime.fromisoformat(created_at)
+
+    now = datetime.datetime.now()
+    now = now.replace(tzinfo=datetime.timezone.utc) # Makes the datetime object timezone-aware
+
+    # If topic is more than two minutes old, don't create an object for it.
+    # This way, we only create objects for topics created after PolicyKit
+    # has been installed to the community.
+    if now - created_at > datetime.timedelta(minutes=2):
+        return False
```

Comment:
This seems like a reasonable approach, but it has a dependency on this value, right?
https://github.com/amyxzhang/policykit/blob/ea3ac7e49078daade578726ec348e0f4c35d27f5/policykit/policykit/settings.py#L238-L239

If the beat schedule is >2m, then we'll miss some posts. maybe add a comment next to each value saying that they are dependent

Reply:
Updated the code -- now the Celery beat schedule time is stored as a constant (CELERY_BEAT_FREQUENCY) in settings.py. discord/tasks.py checks if the topic is more than 2 * CELERY_BEAT_FREQUENCY seconds old and only creates objects for topics that have been created less than that many seconds ago.